### ISSUE TITLE
No topic words in visualization

### ISSUE THREAD

**Issue by sanshine2020** [MaartenGr/BERTopic#88]
Hi Maarten,

I'm calling the BERTopic visualization, but don't see the topic words nor size in the plot. Instead the hoverdata shows customdata[3] and customdata[4], see screenshot. The topic (customdata[2]) however is shown correctly. This issue occurs with the 20 newsgroups dataset as well as with my own data (list of strings). Do you know how to solve this issue?

![image](https://user-images.githubusercontent.com/61740021/112806423-0be50b00-9077-11eb-8cf4-e945c2c0f974.png)

data = fetch_20newsgroups(subset='test', remove=('headers', 'footers', 'quotes'))['data']
topic_model = bertopic.BERTopic()
labels, probabilities = topic_model.fit_transform(data)
fig = topic_model.visualize_topics()
fig.show()

Thnx in advance!

**Comment by MaartenGr** [comment]
Hmmm, I think I have seen this issue before. So far, this has often been caused by an incorrect version of Plotly. I believe you can fix this by installing BERTopic with: `pip install bertopic[visualization]`

Which version of Plotly are you currently using?

**Comment by juliechoong** [comment]
> 
> 
> Hmmm, I think I have seen this issue before. So far, this has often been caused by an incorrect version of Plotly. I believe you can fix this by installing BERTopic with: `pip install bertopic[visualization]`
> 
> Which version of Plotly are you currently using?

I faced the same issue too. This works!

**Comment by MaartenGr** [comment]
Great! Then I will close this for now. The next update will include plotly into the base packages so these issues are unlikely to appear in the future.

**Comment by stepwei** [comment]
Hi Maarten, this error indeed still persists with plotly 5.4.0

**Comment by MaartenGr** [comment]
The dependencies with respect to plotly are currently limited to 4.14.2 as newer versions require a different way of showing words and size. Unfortunately, I do not think I am able to support all versions. In practice, that would mean either `plotly>=4.7.0,<4.14.3` or `plotly>=4.14.3` would be supported. I might opt for the latter if there is enough support to do so and if this does not interfere with common packages.
